.github/workflows/enos-run.yml

```diff
@@ -17,8 +17,49 @@ env:
   PKG_NAME: boundary
 
 jobs:
+  setup:
+    outputs:
+      cache-go-build: ${{ steps.go-cache-paths.outputs.go-build }}
+      cache-go-mod: ${{ steps.go-cache-paths.outputs.go-mod }}
+      cache-go-bin: ${{ steps.go-cache-paths.outputs.go-bin }}
+      go-cache-key: ${{ steps.go-cache-key.outputs.key }}
+    runs-on: ${{ fromJSON(vars.RUNNER) }}
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+        with:
+          fetch-depth: '0'
+      - name: Set up Go
+        uses: actions/setup-go@d0a58c1c4d2b25278816e339b944508c875f3613 # v3.4.0
+        with:
+          go-version: ${{ inputs.go-version }}
+      - name: Determine go cache key
+        id: go-cache-key
+        run: |
+          echo "key=${{ runner.os }}-go-${{ hashFiles('**/go.sum', './Makefile', './tools/tools.go') }}" >> "$GITHUB_OUTPUT"
```

Comment:
Even though this only cares about the Makefile (the version of `tparse`), does it make sense to still codify the cache based on the same files as its used in `test.yml`? That way, it could share those same caches and save on space.

Reply:
Yes, I would make the cache correspond to the things that actually result in a change. So since you are calling `make tools`, any changes to those files would result in a different result when calling `make tools`, even though in this case only tparse is used.